File: spring-core-lab-solutions/Lab-06-solution/petclinic/src/main/java/com/petclinic/core/VisitService.java
package com.petclinic.core;

import org.springframework.stereotype.Service;

@Service
public class VisitService {
    private final VisitRepository visitRepository;

    public VisitService(VisitRepository visitRepository) {
        this.visitRepository = visitRepository;
    }


    public Visit findByReferenceNumber(String referenceNumber) {
        return visitRepository.findByReferenceNumber(referenceNumber);
    }

    public Visit save(Visit visit) {
        return this.visitRepository.save(visit);
    }
}

File: Lab-kotlin/src/main/kotlin/com/petclinic/core/VisitService.kt
package com.petclinic.core

import org.springframework.stereotype.Service

@Service
class VisitService(
        val visitRepository:VisitRepository) {
    fun save(visit: Visit): Visit {
        return visitRepository.save(visit)
    }

    fun findById(id: Long): Visit {
        return visitRepository.findById(id).orElseThrow()
    }
}